lms/static/js/student_account/views/AccessView.js

```diff
@@ -83,7 +83,11 @@ var edx = edx || {};
 
         postRender: function() {
             // Load the default form
-            this.loadForm( this.activeForm );
+            if ( Backbone.history.getHash() == "forgot-password-modal" ) {
```

Comment:
` Backbone.history.getHash()`
This actually checks the page `hash` and matches it with `forgot-password-modal`

If any URL hash matches, then show `forgot password form instead of login form.